test/acceptance/features/steps/steps.py

```diff
@@ -278,6 +279,57 @@ def then_envFrom_contains_intermediate_secret_name(context, intermediate_secret_
         intermediate_secret_name, wait=True, timeout=120) is not None, f"There is no deployment with intermediate secret {intermediate_secret_name}"
 
 
+# STEP
+@given(u'OLM Operator Backend is running')
+def operator_manifest_installed(context):
+    openshift = Openshift()
+    _ = openshift.oc_apply_yaml_file(os.path.join(os.getcwd(), "test/acceptance/resources/backend.operator.manifest.yaml"))
+
+
+# STEP
+@given(u'Backend CR "{cr_name}" is applied')
+def create_backend_cr(context, cr_name):
+    create_cr(context, cr_name)
+
+
+# STEP
+@given(u'Service Binding request is applied')
+def create_sbr(context):
+    sbr_is_applied(context)
+
+
+# STEP
+@when(u'Backend status in "{backend_name}" is updated')
+def backend_status_updated(context, backend_name):
+    create_cr(context, backend_name)
+
+
+# STEP
+@then(u'Secret "{secret_name}" contains "{secret_key}" key with value "{secret_value}"')
+def check_secret_key_value(context, secret_name, secret_key, secret_value):
+    openshift = Openshift()
+    json_path = f'{{.data.{secret_key}}}'
+    value = "true"
+    value_bytes = value.encode('ascii')
+    base64_bytes_value = base64.b64encode(value_bytes)
+    base64_final_value = base64_bytes_value.decode('ascii')
+    output = openshift.get_resource_info_by_jsonpath("secrets", secret_name, context.namespace.name, json_path, wait=True, timeout=30)
+    if secret_value == value and output != base64_final_value:
```

Comment:
It was pointed out a few times on this PR: we should not do polling just if we the step expect value true. The step needs to be generic, because we want to use in other scenarios as well.

Independently from the requested value, we should poll the given secret, read the value under the key and repeat until:

* the request value is read from the secret
* or the given timeout is reached

So, in this case we should have something like:

```python
while True:
  base64_actual_secret_value = openshift.get_resource_info_by_jsonpath("secrets", secret_name, context.namespace.name, json_path, wait=True, timeout=30)
  actual_secret_value = base64.b64decode(base64_actual_secret_value).decode('ascii')
  if secret_value == actual_secret_value:
     break
  // add here repeat logic/checking for timeouts
```

Reply:
Done